modules/user/user.service.js

```diff
@@ -0,0 +1,112 @@
+const { AuthenticationError, UserInputError } = require('apollo-server');
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
+const User = require('./user.model');
+const {
+  validateRegisterInput,
+  validateLoginInput,
+} = require('../../utils/validateUser');
+const generateToken = require('../../utils/createToken');
+const verifyUser = require('../../utils/verifyUser');
+
+class UserService {
+  getAllUsers() {
+    return User.find();
+  }
+
+  getUserById(id, data) {
+    const { token } = data.headers || data.cookies || '';
+    if (!token) {
+      throw new AuthenticationError(JSON.stringify({ message: 'no token' }));
+    }
+
+    if (verifyUser(token)) return User.findById(id);
+  }
+
+  updateUser(id, user) {
+    return User.findByIdAndUpdate(id, user);
+  }
+
+  async loginUser({ email, password }) {
+    const { errors, valid } = validateLoginInput(email, password);
+
+    if (!valid) {
+      throw new UserInputError('Errors', { errors });
+    }
+
+    const user = await User.findOne({ email });
+
+    if (!user) {
+      errors.general = 'User not found';
```

Comment:
Do we need to duplicate the error message `User not found`?

Reply:
fixed

Reply:
I meant that we don't need to pass error message twice